test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs

```diff
@@ -1250,20 +1250,20 @@ public async Task Restore_WithHttpSourceAndFalseAllowInsecureConnections_WarnsCo
             CommandRunnerResult result = RunRestore(pathContext, _successExitCode);
 
             // Assert
-            string formatString = "You are running the 'restore' operation with an 'HTTP' source, '{0}'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS'";
-            string warningForHttpSource = string.Format(formatString, "http://api.source/index.json");
-            string warningForHttpsSource = string.Format(formatString, "https://api.source/index.json");
+            string errorForHttpSource = string.Format(PackageManagement.Strings.Error_HttpSource_Single, "restore", "http://api.source/index.json");
+            string errorForHttpsSource = string.Format(PackageManagement.Strings.Error_HttpSource_Single, "restore", "https://api.source/index.json");
 
             result.Success.Should().BeTrue();
-            Assert.Contains($"Added package 'A.1.0.0' to folder '{projectBPackages}'", result.Output);
-            Assert.DoesNotContain(warningForHttpsSource, result.Output);
-            if (hasHttpWarning)
+            Assert.DoesNotContain(errorForHttpsSource, result.AllOutput);
```

Comment:
Instead of asserting a missing error, why can't this depend on `result.Success.Should().BeTrue()` ?

Reply:
From what I have seen, it is possible for the restore command runner to log an error and still exit code of 0. This is just making sure that is not happening and this error is not logged.